core/lib/zksync_core/src/native_erc20_fetcher/mod.rs

```diff
@@ -0,0 +1,107 @@
+use std::{sync::Arc, time::Duration};
+
+use async_trait::async_trait;
+use metrics::atomics::AtomicU64;
+use tokio::{
+    sync::{watch, OnceCell},
+    task::JoinHandle,
+};
+use zksync_config::configs::native_erc20_fetcher::NativeErc20FetcherConfig;
+
+// TODO: this error type is also defined by the gasAdjuster module,
+//       we should consider moving it to a common place
+#[derive(thiserror::Error, Debug, Clone)]
+#[error(transparent)]
+pub struct Error(Arc<anyhow::Error>);
+
+impl From<anyhow::Error> for Error {
+    fn from(err: anyhow::Error) -> Self {
+        Self(Arc::new(err))
+    }
+}
+
+pub trait Erc20Fetcher: 'static + std::fmt::Debug + Send + Sync {
```

Comment:
I changed the trait's name (and of other structs as well) to better reflect their meaning and also to use the word `NativeToken` instead of `NativeErc20` here [6838588](https://github.com/matter-labs/zksync-era/pull/996/commits/6838588290160de7db1b09d7bcaf168a28366b70), as the native token could be `eth` or maybe in the other future a non-erc20 as well. I also added some small comment/doc to the trait.

Reply:
That looks better, thanks!